lib/Model/ModelInterface.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Model;
+
+interface ModelInterface
+{
+    /**
+     * @return Dao\AbstractDao
+     */
+    public function getDao();
+
+    /**
+     * @param Dao\AbstractDao $dao
+     *
+     * @return self
+     */
+    public function setDao($dao);
+
+    /**
+     * @param string|null $key
+     * @param bool $forceDetection
+     *
+     * @throws \Exception
+     */
+    public function initDao($key = null, $forceDetection = false);
+
+    /**
+     * @param string $modelClass
+     *
+     * @return string|null
+     */
+    public static function locateDaoClass($modelClass);
```

Comment:
Not really needed on the interface, right?

Reply:
You are right. I removed it.